src/main/java/com/fasterxml/jackson/databind/util/EnumResolver.java

```diff
@@ -45,7 +52,29 @@ public static EnumResolver constructFor(Class<Enum<?>> enumCls, AnnotationIntros
             }
             map.put(name, enumValues[i]);
         }
-        return new EnumResolver(enumCls, enumValues, map);
+
+        Enum<?> defaultEnum = determineDefaultEnumValue(enumCls);
+
+        return new EnumResolver(enumCls, enumValues, map, defaultEnum);
+    }
+
+    private static Enum<?> determineDefaultEnumValue(Class<Enum<?>> enumCls) {
+        Field[] fields = ClassUtil.getDeclaredFields(enumCls);
+        for (Field field : fields) {
+            JsonEnumDefaultValue defaultValueAnnotation = field.getAnnotation(JsonEnumDefaultValue.class);
+            if (defaultValueAnnotation != null && field.isEnumConstant()) {
+                try {
+                    Method valueOf = enumCls.getDeclaredMethod("valueOf", String.class);    // using `getMethod` causes IllegalAccessException
+                    valueOf.setAccessible(true);
+                    return enumCls.cast(valueOf.invoke(null, field.getName()));
+                } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+                    java.util.logging.Logger.getLogger(EnumResolver.class.getName())
```

Comment:
@cowtowncoder based on your comment on logging, i'm not sure if it would it be best to wrap this into an exception and let it bubble up or just swallow it and return null?